.github/workflows/build-and-test.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# © 2021-2023 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+# Install Simics Base
+PACKAGE_REPO=https://af02p-or.devtools.intel.com/artifactory/simics-repos/pub/simics-6/linux64/
+INSTALL_DIR=install
+ispm install 1000-6.latest -y --install-dir $INSTALL_DIR --package-repo $PACKAGE_REPO
+if [ $? -ne 0 ]; then
+  echo "*** Failed to install Simics Base, see build-log for details"
+  exit 1
+fi
+
+# Create project
+DMLC_PROJ=proj
+mkdir -p $DMLC_PROJ
+./$INSTALL_DIR/*/bin/project-setup $DMLC_PROJ
+pushd $DMLC_PROJ
+
+# Setup link to DMLC sources in repo
+pushd modules/
+ln -s ../.. dmlc
+popd
+
+# Local build of dmlc
+CPUS=$(cat /proc/cpuinfo | grep ^processor | wc -l)
+make dmlc -j $CPUS || exit $?
+
+# Run the tests
+T126_JOBS=$CPUS DMLC_CC=/usr/itm/gcc/13.1.0/bin/gcc ./bin/test-runner --moduledirs modules/dmlc --suite modules/dmlc/test || exit $?
```

Comment:
Some paths are hard-coded to our env. Would it make sense to start with some of the paths overrideable, like
```
if [ -n "$(DMLC_CC)" ] ; then export DMLC_CC=/usr/itm/... ; fi
```
... and probably similar for `PACKAGE_REPO`?
We can do the same later on request, of course.

Reply:
Yes, this is just a quick hack to get the box moved from TC into GH ; we`re still running (or I was assuming we would still be running) on our local runners where we have simicsmp adn /usr/itm/gcc . 

If we want to run outside on generic runners, we must somehow package up a docker image (or an image diff that you apply to some standard ubuntu image) that provides a decent GCC and Simics Base. I don't know how to do that, so that will have to be done and resolved later.